MAINT: Remove geopandas.datasets

m-richards opened the pull request:
Closes #3079 
Closes #2751

martinfleis commented on geopandas/datasets/__init__.py:
```diff
@@ -1,59 +1,25 @@
-        raise ValueError(msg)
+        error_msg = (
+            "The geopandas.dataset has been deprecated and "
+            "was removed in GeoPandas 1.0. Sample datasets are now available "
+            "in the geodatasets package (https://geodatasets.readthedocs.io/en/latest/)"
```
Only some are there. You will not find `naturalearth_lowres` in there for example. We may want to be more careful with phrasing here.

martinfleis commented on geopandas/datasets/__init__.py:
```diff
@@ -1,59 +1,25 @@
-        msg = "The dataset '{data}' is not available. ".format(data=dataset)
-        msg += "Available datasets are {}".format(", ".join(available))
-        raise ValueError(msg)
+        error_msg = (
```
Yeah, I think that makes sense.

martinfleis commented on geopandas/tests/util.py:
```diff
@@ -13,6 +13,15 @@
 PACKAGE_DIR = os.path.dirname(os.path.dirname(HERE))
 
+_TEST_DATA_DIR = os.path.join(PACKAGE_DIR, "geopandas", "tests", "data")
+_NYBB = "zip://" + os.path.join(_TEST_DATA_DIR, "nybb_16a.zip")
+_NATURALEARTH_CITIES = os.path.join(
+    _TEST_DATA_DIR, "naturalearth_cities", "naturalearth_cities.shp"
+)
+_NATURALEARTH_LOWRES = os.path.join(
+    _TEST_DATA_DIR, "naturalearth_lowres", "naturalearth_lowres.shp"
+)
```
I suppose we can use geodatasets in benchmarks to avoid this.

m-richards replied:
As long as the caching in geodatasets won't mess with the benchmarks? I imagine it would be fine once each dataset exists in the local cache, just that there might be a small slowdown in the benchmarks after this PR?

martinfleis reviewed (commented):
Thanks a lot for looking into that! 

I suppose that the files won't be shipped in the wheel, correct?

m-richards commented:
> Thanks a lot for looking into that!
> 
> I suppose that the files won't be shipped in the wheel, correct?

At present, for this PR nybb is included, and naturalearth is excluded - that's not by design, it's just that I haven't updated the existing rules in `pyproject.toml::[tool.setuptools.package-data]`. I'm not sure what the "right thing" is to do here - I could see the argument we should include these so the test suite in the wheel/sdist works, but I could also see an argument that directing a downstream consumer to github for this would be fine too. I don't know but I'd imagine system linux package managers are working with source code from github rather than sdists/ wheels from pypi.

m-richards commented on benchmarks/clip.py:
```diff
@@ -1,11 +1,13 @@
+from geopandas import read_file, clip
 from shapely.geometry import box
+from geopandas.tests.util import _NATURALEARTH_LOWRES
+from geopandas.tests.util import _NATURALEARTH_CITIES
```
I take it then you don't run these benchmarks by checking out 1.0 / 0.14 and then running this script on each (which would be different for each, and so shouldn't need a try/except), but rather take the 1.0 version of this script and run it against a pip installed version of 0.14/1.0?

jorisvandenbossche replied:
That's indeed how it more or less works, yes (`asv` handles that part for you)

m-richards commented:
I think this is ready to go now - contrary to what I said at the meeting today, I had already updated the pyproject.toml so that datasets are excluded from the wheel. I've built it locally and confirmed they aren't present. 

I went ahead and added pytest skips into the fixtures, I've that the skips work, and also that they do not skip when run on github actions: https://github.com/geopandas/geopandas/actions/runs/7415769532/job/20179553948?pr=3124

m-richards commented:
In its current form, I think this also resolves #2751

martinfleis reviewed (approved):
Thanks a lot! I think this is ready. Can you just add a changelog note?

m-richards commented:
> Thanks a lot! I think this is ready. Can you just add a changelog note?

Whoops, forgot to push it yesterday, should be there now.

jorisvandenbossche commented:
Thanks Matt!
